aclk/aclk_query.c

```diff
@@ -111,6 +111,15 @@ static int http_api_v2(struct aclk_query_thread *query_thr, aclk_query_t query)
     w->tv_in = query->created_tv;
     now_realtime_timeval(&w->tv_ready);
 
+    if (query->timeout && (dt_usec(&query->created_tv, &w->tv_ready) / 1000.0) > query->timeout) {
```

Comment:
I am thinking should this be after we check about the NODE_ID validity?
If cloud sends query with malformed or wrong node_id perhaps that specific error should be sent despite timeout?

Reply:
We can improve with a followup PR a few things (for example the buffers as it will not use them to generate a response). The timeout basically means that even the node check with the specific error may not reach the cloud (because of the timeout) so no one is listening there anymore.